zeppelin-server/src/main/java/org/apache/zeppelin/rest/CredentialRestApi.java

```diff
@@ -63,10 +63,15 @@ public CredentialRestApi(Credentials credentials) {
   @PUT
   public Response putCredentials(String message) throws IOException {
     Map<String, String> messageMap = gson.fromJson(message,
-            new TypeToken<Map<String, String>>(){}.getType());
+      new TypeToken<Map<String, String>>(){}.getType());
     String entity = messageMap.get("entity");
     String username = messageMap.get("username");
     String password = messageMap.get("password");
+
+    if (entity == null || username == null || password == null) {
```

Comment:
An empty string does not have a problem in Zeppelin server.
I only solve a serious trouble in this pr.